File: client/src/shared/models/chat/MessageDTO.ts
import { MessageFileDTO } from "./MessageFileDTO";
import { MessageType } from "./MessageType";
import { SenderDTO } from "./SenderDTO";

export interface MessageDTO {
    id:number;
    content:string;
    sender:SenderDTO;
    time:string;
    type: string;
    chatId:number;
    files:MessageFileDTO[];
}

File: server/src/main/java/com/project/WebSockets/dto/MessageDTO.java
package com.project.WebSockets.dto;

import com.project.WebSockets.models.MessageType;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class MessageDTO {
    private Long id;
    private String content;
    private SenderDTO sender;
    private LocalDateTime time;
    private MessageType type;
    private Long chatId;
    private List<MessageFileDTO> files;
}
